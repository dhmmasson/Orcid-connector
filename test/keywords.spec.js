const {
  getOrcidKeywords,
  getPapersAndKeywords,
  translateKeywords,
  cleanKeywords,
} = require("../src/hal");
// Check that the getOrcidKeywords function returns the expected keywords

describe("getOrcidKeywords", () => {
  it("should return the expected keywords", async () => {
    const keywords = await getOrcidKeywords("0000-0002-7072-3146");
    // expect the result to be an array of strings to be returned
    expect(keywords).toBeInstanceOf(Array);
  });
});

describe("translateKeywords", () => {
  it("should return the expected keywords", async () => {
    const keywords = await translateKeywords(["humain", "créativité"]);
    // expect the result to be an array of strings to be returned
    expect(keywords).toEqual(["human", "creativity"]);
  });
});

// Check that the getPapersAndKeywords function returns the expected papers and keywords
describe("getPapersAndKeywords", () => {
  it("should return the expected papers and keywords", async () => {
    const keywords = await getPapersAndKeywords("dimitri-masson");
    // expect the result to be an array of objects to be returned
    expect(keywords).toBeInstanceOf(Array);
    // Snapshot testing
    expect(keywords).toMatchSnapshot();
  });
});

describe("Snapshot Keywords", () => {
  it("Raw", async () => {
    const keywords = await getPapersAndKeywords("dimitri-masson");
    expect([...new Set(keywords)].sort()).toMatchSnapshot();
  });
  it("Cleaned", async () => {
    const keywords = await getPapersAndKeywords("dimitri-masson");
    const cleanedKeywords = await cleanKeywords(keywords);
    expect(cleanedKeywords).toMatchSnapshot();
  });
});

const keywords = [
  "ALTERNATIVES RANKING",
  "BRAINSTORMING",
  "BUSINESS PROCESS MODELING",
  "CIRCULARITY",
  "CLUSTER",
  "COLLABORATIVE DESIGN",
  "COLLECTIVE DECISION",
  "COMPETITIVENESS CLUSTER",
  "CONCEPT GENERATION",
  "CREATIVE SUPPORT TOOL",
  "CREATIVE TEAMS",
  "CREATIVITY",
  "CREATIVITY -- EVALUATION",
  "CYBER-PHYSICAL AND HUMAN SYSTEMS",
  "CYBER-PHYSICAL SYSTEMS AND HUMANS",
  "DECISION SUPPORT",
  "DECISION SUPPORT SYSTEM",
  "DECISION SUPPORT SYSTEM DSS",
  "DECISION UNDER UNCERTAINTY",
  "DESIGN",
  "DESIGN FOR RECYCLING",
  "DESIGN GALLERIES",
  "DESIGN METHOD",
  "DESIGN SPACES",
  "DESIGN THINKING",
  "DIGITAL PRODUCT PASSPORT",
  "DIVERSITY",
  "DYNAMIC USER INTERFACE UI",
  "EMOTIONAL DESIGN",
  "EVALUATION",
  "EVOLUTIONARY ALGORITHM",
  "EVOLUTIONARY ALGORITHMS",
  "EXAMPLE",
  "EXAMPLES GALLERIES",
  "EXPLORATION",
  "FASHION INDUSTRY",
  "GALLERY",
  "GALLERY OF EXAMPLES",
  "GENERATION",
  "GRAPH OF MODELS",
  "HACKATHON",
  "HETEROGENEOUS DATA",
  "HUMAN-COMPUTER INTERFACE",
  "HUMAN-COMPUTER INTERFACE DESIGN",
  "IDEA GENERATION",
  "IDEATION",
  "INNOVATION",
  "INSPIRATION",
  "INTERACTIVE GENETIC ALGORITHMS",
  "INTERCLUSTERING",
  "INTEROPERABILITY",
  "MANAGEMENT",
  "MANUFACTURING SYSTEM ENGINEERING",
  "MATCHING",
  "MATERIAL RECOVERY",
  "METHODOLOGY",
  "MODEL BASED USER INTERFACES",
  "MULTI CRITERIA DECISION MAKING",
  "MULTI-CRITERIA DECISION-MAKING",
  "NEAR ZERO ENERGY BUILDING REFURBISHMENT",
  "NEW PRODUCT DEVELOPMENT",
  "PRECISION BREEDING",
  "PRECISION LIVESTOCK FARMING",
  "PRODUCT DISMANTLING",
  "PRODUCT TRANSPARENCY",
  "RATING",
  "SENSITIVITY ANALYSIS",
  "SKETCHING",
  "STAR",
  "STUDENTS",
  "SUPPLY CHAIN",
  "SUSTAINABLE DECISIONS",
  "TANGIBLE INTERFACE",
  "TEXTILE",
  "TEXTILES AND CLOTHING",
  "THE 24H OF INNOVATION",
  "TOOL",
  "TOOL TO SUPPORT CREATIVITY",
  "TRACEABILITY",
  "TRANSPARENCY",
  "USER CENTERED DESIGN",
  "USER INTERFACE DESIGN",
  "UX",
  "VIRTUAL REALITY",
];

describe("debug Test", () => {
  it("should return the expected keywords", async () => {
    let keywords = await getPapersAndKeywords("dimitri-masson");
    keywords = keywords.slice(20, 30);
    console.table(keywords);
    const cleanedKeywords = await cleanKeywords(keywords);
    console.table(cleanedKeywords);
    keywords = await getPapersAndKeywords("dimitri-masson");
    // expect true to be true
    expect(true).toBe(true);
  });
});

