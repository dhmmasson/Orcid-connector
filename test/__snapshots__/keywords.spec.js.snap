// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Keywords Cleaned 1`] = `
[
  "ALTERNATIVES RANKING",
  "BRAINSTORMING",
  "BUSINESS PROCESS MODELING",
  "CIRCULARITY",
  "CLUSTER",
  "COLLABORATIVE DESIGN",
  "COLLECTIVE DECISION",
  "COMPETITIVENESS CLUSTER",
  "CONCEPT GENERATION",
  "CREATIVE SUPPORT TOOL",
  "CREATIVE TEAMS",
  "CREATIVITY",
  "CREATIVITY -- EVALUATION",
  "CYBER-PHYSICAL AND HUMAN SYSTEMS",
  "CYBER-PHYSICAL SYSTEMS AND HUMANS",
  "DECISION SUPPORT",
  "DECISION SUPPORT SYSTEM",
  "DECISION SUPPORT SYSTEM DSS",
  "DECISION UNDER UNCERTAINTY",
  "DESIGN",
  "DESIGN FOR RECYCLING",
  "DESIGN GALLERIES",
  "DESIGN METHOD",
  "DESIGN SPACES",
  "DESIGN THINKING",
  "DIGITAL PRODUCT PASSPORT",
  "DIVERSITY",
  "DYNAMIC USER INTERFACE UI",
  "EMOTIONAL DESIGN",
  "EVALUATION",
  "EVOLUTIONARY ALGORITHM",
  "EVOLUTIONARY ALGORITHMS",
  "EXAMPLE",
  "EXAMPLES GALLERIES",
  "EXPLORATION",
  "FASHION INDUSTRY",
  "GALLERY",
  "GALLERY OF EXAMPLES",
  "GENERATION",
  "GRAPH OF MODELS",
  "HACKATHON",
  "HETEROGENEOUS DATA",
  "HUMAN-COMPUTER INTERFACE",
  "HUMAN-COMPUTER INTERFACE DESIGN",
  "IDEA GENERATION",
  "IDEATION",
  "INNOVATION",
  "INSPIRATION",
  "INTERACTIVE GENETIC ALGORITHMS",
  "INTERCLUSTERING",
  "INTEROPERABILITY",
  "MANAGEMENT",
  "MANUFACTURING SYSTEM ENGINEERING",
  "MATCHING",
  "MATERIAL RECOVERY",
  "METHODOLOGY",
  "MODEL BASED USER INTERFACES",
  "MULTI CRITERIA DECISION MAKING",
  "MULTI-CRITERIA DECISION-MAKING",
  "NEAR ZERO ENERGY BUILDING REFURBISHMENT",
  "NEW PRODUCT DEVELOPMENT",
  "PRECISION BREEDING",
  "PRECISION LIVESTOCK FARMING",
  "PRODUCT DISMANTLING",
  "PRODUCT TRANSPARENCY",
  "RATING",
  "SENSITIVITY ANALYSIS",
  "SKETCHING",
  "STAR",
  "STUDENTS",
  "SUPPLY CHAIN",
  "SUSTAINABLE DECISIONS",
  "TANGIBLE INTERFACE",
  "TEXTILE",
  "TEXTILES AND CLOTHING",
  "THE 24H OF INNOVATION",
  "TOOL",
  "TOOL TO SUPPORT CREATIVITY",
  "TRACEABILITY",
  "TRANSPARENCY",
  "USER CENTERED DESIGN",
  "USER INTERFACE DESIGN",
  "UX",
  "VIRTUAL REALITY",
]
`;

exports[`Snapshot Keywords Raw 1`] = `
[
  "Aide à la décision",
  "Algorithmes évolutionnaires",
  "Alternatives ranking",
  "Brainstorming",
  "Business Process Modeling",
  "Circularity",
  "Cluster",
  "Collaborative design",
  "Collective Decision",
  "Concept generation",
  "Conception",
  "Conception d’Interfaces Homme-Machine",
  "Creative support tool",
  "Creative teams",
  "Creativity",
  "Créativité",
  "Créativité -- évaluation",
  "Cyber-physical and human systems",
  "Decision Support System",
  "Decision Support System DSS",
  "Decision support system",
  "Decision under uncertainty",
  "Design Thinking",
  "Design for recycling",
  "Design galleries",
  "Design method",
  "Design spaces",
  "Diversity",
  "Données hétérogènes",
  "Dynamic User Interface UI",
  "Décision collective",
  "Emotional design",
  "Evaluation",
  "Evolutionary algorithm",
  "Examples galleries",
  "Exemple",
  "Exploration",
  "Fashion industry",
  "Galerie",
  "Galerie d'exemples",
  "Generation",
  "Graph of models",
  "Génération",
  "Hackathon",
  "Heterogeneous data",
  "Idea generation",
  "Ideation",
  "Innovation",
  "Inspiration",
  "Interactive Genetic Algorithms",
  "Interclustering",
  "Interface Homme-Machine",
  "Interoperability",
  "Interopérabilité",
  "Management",
  "Manufacturing System Engineering",
  "Matching",
  "Material recovery",
  "Methodology",
  "Model based User Interfaces",
  "Multi criteria decision making",
  "Multi-Criteria Decision-MakingMCDM",
  "Near Zero Energy Building Refurbishment",
  "Near Zero Energy Building refurbishment",
  "New product development",
  "Outil support à la créativité",
  "Passeport produit Numérique",
  "Precision livestock farming",
  "Product dismantling",
  "Product transparency",
  "Pôle de compétivivité",
  "Rating",
  "Réalité virtuelle",
  "STAR",
  "Sensitivity analysis",
  "Sketching",
  "Students",
  "Supply chain",
  "Sustainable Decisions",
  "Sustainable decisions",
  "Systèmes cyber-physiques et humains",
  "Tangible interface",
  "Textile",
  "Textiles and Clothing",
  "The 24h of innovation",
  "Tool",
  "Traceability",
  "Transparence",
  "Transparency",
  "Traçabilité",
  "UX",
  "User Interface Design",
  "User Interface design",
  "User centred design",
  "Virtual Reality",
  "Élevage de précision",
]
`;

exports[`debug Test should return the expected keywords 1`] = `
[
  "Aide à la décision",
  "Algorithmes évolutionnaires",
  "Alternatives ranking",
  "Brainstorming",
  "Business Process Modeling",
  "Circularity",
  "Cluster",
  "Collaborative design",
  "Collective Decision",
  "Concept generation",
  "Conception",
  "Conception d’Interfaces Homme-Machine",
  "Creative support tool",
  "Creative teams",
  "Creativity",
  "Créativité",
  "Créativité -- évaluation",
  "Cyber-physical and human systems",
  "Decision Support System",
  "Decision Support System DSS",
  "Decision support system",
  "Decision under uncertainty",
  "Design Thinking",
  "Design for recycling",
  "Design galleries",
  "Design method",
  "Design spaces",
  "Diversity",
  "Données hétérogènes",
  "Dynamic User Interface UI",
  "Décision collective",
  "Emotional design",
  "Evaluation",
  "Evolutionary algorithm",
  "Examples galleries",
  "Exemple",
  "Exploration",
  "Fashion industry",
  "Galerie",
  "Galerie d'exemples",
  "Generation",
  "Graph of models",
  "Génération",
  "Hackathon",
  "Heterogeneous data",
  "Idea generation",
  "Ideation",
  "Innovation",
  "Inspiration",
  "Interactive Genetic Algorithms",
  "Interclustering",
  "Interface Homme-Machine",
  "Interoperability",
  "Interopérabilité",
  "Management",
  "Manufacturing System Engineering",
  "Matching",
  "Material recovery",
  "Methodology",
  "Model based User Interfaces",
  "Multi criteria decision making",
  "Multi-Criteria Decision-MakingMCDM",
  "Near Zero Energy Building Refurbishment",
  "Near Zero Energy Building refurbishment",
  "New product development",
  "Outil support à la créativité",
  "Passeport produit Numérique",
  "Precision livestock farming",
  "Product dismantling",
  "Product transparency",
  "Pôle de compétivivité",
  "Rating",
  "Réalité virtuelle",
  "STAR",
  "Sensitivity analysis",
  "Sketching",
  "Students",
  "Supply chain",
  "Sustainable Decisions",
  "Sustainable decisions",
  "Systèmes cyber-physiques et humains",
  "Tangible interface",
  "Textile",
  "Textiles and Clothing",
  "The 24h of innovation",
  "Tool",
  "Traceability",
  "Transparence",
  "Transparency",
  "Traçabilité",
  "UX",
  "User Interface Design",
  "User Interface design",
  "User centred design",
  "Virtual Reality",
  "Élevage de précision",
]
`;

exports[`getPapersAndKeywords should return the expected papers and keywords 1`] = `
[
  "Precision livestock farming",
  "Decision support system",
  "Collective Decision",
  "Interoperability",
  "Heterogeneous data",
  "Traceability",
  "Transparency",
  "Fashion industry",
  "Supply chain",
  "Product transparency",
  "Traceability",
  "Transparency",
  "Textiles and Clothing",
  "Fashion industry",
  "Supply chain",
  "Brainstorming",
  "User Interface Design",
  "Creativity",
  "Evolutionary algorithm",
  "Design galleries",
  "Conception d’Interfaces Homme-Machine",
  "Créativité",
  "Algorithmes évolutionnaires",
  "Galerie d'exemples",
  "Sustainable decisions",
  "Decision Support System",
  "Alternatives ranking",
  "Near Zero Energy Building refurbishment",
  "Multi criteria decision making",
  "Matching",
  "User centred design",
  "Emotional design",
  "New product development",
  "Creativity",
  "Design Thinking",
  "Innovation",
  "Hackathon",
  "The 24h of innovation",
  "Diversity",
  "Creative teams",
  "Students",
  "Manufacturing System Engineering",
  "Decision Support System",
  "Business Process Modeling",
  "Creativity",
  "Tangible interface",
  "Ideation",
  "Idea generation",
  "Creativity",
  "Concept generation",
  "Virtual Reality",
  "Sketching",
  "Réalité virtuelle",
  "Créativité",
  "UX",
  "Near Zero Energy Building Refurbishment",
  "Multi criteria decision making",
  "Alternatives ranking",
  "Sustainable Decisions",
  "Decision Support System",
  "Interface Homme-Machine",
  "Conception",
  "Créativité",
  "Inspiration",
  "Exemple",
  "Exploration",
  "Galerie",
  "Génération",
  "Innovation",
  "Evaluation",
  "Cluster",
  "Collaborative design",
  "Creativity",
  "Creativity",
  "Créativité -- évaluation",
  "Creativity",
  "Innovation",
  "Methodology",
  "STAR",
  "Tool",
  "Management",
  "Creative support tool",
  "Creativity",
  "Innovation",
  "Ideation",
  "Evaluation",
  "Rating",
  "Outil support à la créativité",
  "Créativité",
  "Creativity",
  "Rating",
  "Créativité",
  "Innovation",
  "Evaluation",
  "Cluster",
  "Pôle de compétivivité",
  "Interclustering",
  "Model based User Interfaces",
  "Graph of models",
  "Design spaces",
  "Creativity",
  "Creativity",
  "User Interface design",
  "Examples galleries",
  "Generation",
  "Interactive Genetic Algorithms",
  "Circularity",
  "Design method",
  "Design for recycling",
  "Material recovery",
  "Product dismantling",
  "Traçabilité",
  "Transparence",
  "Passeport produit Numérique",
  "Textile",
  "Decision Support System DSS",
  "Multi-Criteria Decision-MakingMCDM",
  "Dynamic User Interface UI",
  "Sensitivity analysis",
  "Decision under uncertainty",
  "Precision livestock farming",
  "Decision support system",
  "Collective Decision",
  "Interoperability",
  "Heterogeneous data",
  "Cyber-physical and human systems",
  "Aide à la décision",
  "Élevage de précision",
  "Systèmes cyber-physiques et humains",
  "Décision collective",
  "Données hétérogènes",
  "Interopérabilité",
  "Traceability",
  "Transparency",
  "Fashion industry",
  "Supply chain",
  "Product transparency",
]
`;
